src/Generators/ModelGenerator.php

```diff
@@ -18,16 +19,55 @@ class ModelGenerator extends BaseGenerator
     /** @var string */
     private $fileName;
 
+    /** @var Doctrine\DBAL\Schema\SchemaManager */
+    private $schemaManager;
+
+    /** @var string */
+    private $table;
+
+    /** @var array */
+    private $tables;
+
+    /** @var array */
+    private $prep;
+
+    /** @var array */
+    private $eloquentRules;
+    /**
+     * @var Doctrine\DBAL\Connection
+     */
+    private $connection;
+
+    /**
+     * Fields not included in the generator by default.
+     *
+     * @var array
+     */
+    protected $excluded_fields = [
+      'created_at',
+      'updated_at',
+    ];
+
+    /**
+     * ModelGenerator constructor.
+     *
+     * @param \InfyOm\Generator\Common\CommandData $commandData
+     */
     public function __construct(CommandData $commandData)
     {
         $this->commandData = $commandData;
         $this->path = $commandData->config->pathModel;
         $this->fileName = $this->commandData->modelName.'.php';
+        $this->table = $this->commandData->dynamicVars['$TABLE_NAME$'];
+        $this->getSchemaManager();
+        $this->getTables();
+        $this->getColumnsPrimaryAndForeignKeysPerTable();
+        $this->getEloquentRules();
     }
 
     public function generate()
     {
-        $templateData = TemplateUtil::getTemplate('model', 'laravel-generator');
+        $templateData = TemplateUtil::getTemplate('models.model', 'laravel-generator');
```

Comment:
Unsure how you want to handle this as users who use published will need to react to new .stub file scructure